### PR TITLE
SG-35936 support for houdini 20.5

### PR DESCRIPTION
**Pull request by eduardoChaucaGallegos**
- Support for houdini 20.5

### CODE REVIEW

**Review by carlos-villavicencio-adsk — APPROVED** [review]
LGTM.

File: plugins/basic/python3.11libs/pythonrc.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2017 Shotgun Software Inc.
```
Comment (julien-lang) [inline plugins/basic/python3.11libs/pythonrc.py:1]: Wrong year

**Review by julien-lang — APPROVED** [review]
Did you manage to understand why we need to add these files for every new Python version?

**Comment by eduardoChaucaGallegos** [comment]
> Did you manage to understand why we need to add these files for every new Python version?

These Python RC files are called by Houdini with `MOT_Director::execPythonRCFiles`. So the new version of Houdini 20.5.278 uses Python 11 and tries to find the RC files in tk-houdini for Python 11

**Comment by carlos-villavicencio-adsk** [comment]
It's seems a bit detailed in https://github.com/shotgunsoftware/tk-houdini/blob/master/plugins/README.md and https://www.sidefx.com/docs/houdini/hom/locations

**Comment by brettmaker** [comment]
I tried this branch and seem to be getting a segmentation fault on launch about 95% of the time. This was with Houdini 20.5.278 Python 3.11 on Linux (Rocky 8). However the Python 3.10 of Houdini 20.5.278 doesn't crash. Has anyone else experienced this issue? Not sure what's causing it.

**Comment by MoisMoshev** [comment]
I also get these crashes. I suspect they are not directly related to the changes here. There is discussion here https://community.shotgridsoftware.com/t/tk-houdini-engine-for-houdini-20/18082/14
where the suspicion is that QApplication is not instantiated correctly.

**Comment by brettmaker** [comment]
> Hi @brettmaker, I had that problem on Windows, so repairing or reinstalling Houdini worked for me.

I haven't tried it on Windows yet. For Linux I tried the production build and the latest daily build of Houdini 20.5 (Python 3.11). Other plugins like HtoA seemed to load fine but when I launched through the toolkit environment then it would crash.

I can confirm the Houdini 20.5 Python 3.10 version doesn't crash with the toolkit environment and that's the same code, so it's something with Python 3.11.

**Comment by MoisMoshev** [comment]
Here is a stack trace. Apparently it happens on executing some Python. On Windows the case seems similar.
[crash.mmoshev_bottleship.local_827622_log.log](https://github.com/user-attachments/files/16715576/crash.mmoshev_bottleship.local_827622_log.log)

**Comment by eduardoChaucaGallegos** [comment]
@MoisMoshev and @brettmaker, I was testing in a Linux machine and I found the same crash issue, mostly of the time crashes 
exist in version 20.5.278 and crashes don't exist in version 20.5.332. I also tested on Windows getting the same result.

So I recommend using the version 20.5.332 with this new PR support

**Comment by brettmaker** [comment]
Unfortunately I didn't have much luck with 20.5.332 on Rocky 8. Attached python traceback and crash log.
[crash.brett_294509_log.txt](https://github.com/user-attachments/files/16806345/crash.brett_294509_log.txt)
[python-traceback.txt](https://github.com/user-attachments/files/16806361/python-traceback.txt)

**Comment by dovanbel** [comment]
Hi,
For me this PR works when used with Houdini 20.5.332 on Windows (py 3.11)

However, when also loading HtoA htoa-6.3.4.0_r0d5c67f_houdini-20.5.332 I get a crash while launching Houdini :  

```
Traceback from 3024 ThreadId=0x000047a8
CURRENT THREAD 18344
+0x7ffaac20ff8f [PySideSignalTypeF] C:\PROGRA~1\SIDEEF~1\HOUDIN~1.332\python311\lib\site-packages-forced\PySide2\pyside2.cp311-win_amd64.dll
+0x7ffb10c252b1 [PyObject_MakeTpCall] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c25511 [PyObject_Vectorcall] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10d2834d [PyEval_EvalFrameDefault] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10d2b83e [PyEval_EvalFrameDefault] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10d1cdf0 [PyWarnings_Init] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c6cc98 [PyCFunction_GetFlags] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c24e99 [PyBytes_Repeat] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c25511 [PyObject_Vectorcall] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10d2834d [PyEval_EvalFrameDefault] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10d2b83e [PyEval_EvalFrameDefault] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c2573d [PyFunction_Vectorcall] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c250d4 [PyObject_FastCallDictTstate] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c259c5 [PyObject_Call_Prepend] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c964ac [PyType_Ready] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c89d5f [PyType_Name] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c252b1 [PyObject_MakeTpCall] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10c25511 [PyObject_Vectorcall] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10d2834d [PyEval_EvalFrameDefault] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x7ffb10d23861 [PyEval_EvalCode] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\python311.dll
+0x290370bd [PY_CompiledCode::evaluateUsingDicts] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\libPY.dll
+0x29036edd [PY_CompiledCode::evaluate] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\libPY.dll
+0x29045ad3 [PYrunPythonStatementsFromFile] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\libPY.dll
+0x1466c07e [MOT_Director::execPythonRCFiles] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\libMOT.dll
+0x146678af [MOT_Director::MOT_Director] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\libMOT.dll
+0x138ff046 [OPUI_MainApp::initApplication] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\libOPUI.dll
+0x112adac2 [AP_Interface::loadWindowGeometry] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\libSI.dll
+0x112ae7d5 [myWinMain] C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\libSI.dll
+0x1400015a2 C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\houdini.exe
+0x140001a32 C:\Program Files\Side Effects Software\Houdini 20.5.332\bin\houdini.exe
+0x7ffb89f3257d [BaseThreadInitThunk] C:\Windows\System32\KERNEL32.DLL
+0x7ffb8ad6af28 [RtlUserThreadStart] C:\Windows\SYSTEM32\ntdll.dll
```

**Comment by brettmaker** [comment]
Check out this sorcery. Just calling QApplication.instance() causes python to bug out.
```
        for i in range(5):
            print("i =", i)

        print("INSTANCE")
        if not QtGui.QApplication.instance():
            # there is a QApplication, so we can load fonts.
            return

        for i in range(5):
            print("i =", i)
```
Output
```
i = 0
i = 1
i = 2
i = 3
i = 4
INSTANCE
i = 0
i = 140316086984888
i = 280632173969776
i = 420948260954664
i = 561264347939552
```
and then eventually:
```
23088: Fatal error: Segmentation fault (sent by pid 8)
```
This is more of an issue with tk-core and the PySide2 patcher. posting findings here for now

**Comment by MoisMoshev** [comment]
> Check out this sorcery. Just calling QApplication.instance() causes python to bug out.
> 
> ```
>         for i in range(5):
>             print("i =", i)
> 
>         print("INSTANCE")
>         if not QtGui.QApplication.instance():
>             # there is a QApplication, so we can load fonts.
>             return
> 
>         for i in range(5):
>             print("i =", i)
> ```
> 
> Output
> 
> ```
> i = 0
> i = 1
> i = 2
> i = 3
> i = 4
> INSTANCE
> i = 0
> i = 140316086984888
> i = 280632173969776
> i = 420948260954664
> i = 561264347939552
> ```
> 
> and then eventually:
> 
> ```
> 23088: Fatal error: Segmentation fault (sent by pid 8)
> ```
> 
> This is more of an issue with tk-core and the PySide2 patcher. posting findings here for now

So should this be reproducible in the tk-core tests of qt loader? I'll give it a try.

**Comment by MoisMoshev** [comment]
We contacted SideFX support, and they said they will prepare a fix.

**Comment by MoisMoshev** [comment]
We tested and it works! I am still getting the lxml error on one machine, that seems like a different issue.

**Comment by eduardoChaucaGallegos** [comment]
> We tested and it works! I am still getting the lxml error on one machine, that seems like a different issue.

That's correct! the daily build (20.5.368) that SideFX released works fine with this PR

**Comment by carlos-villavicencio-adsk** [comment]
Did you check the release notes? This one looks interesting

image

**Pull request merged**